app/Libraries/Score/UserRank.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace App\Libraries\Score;
+
+use App\Libraries\Search\ScoreSearch;
+use App\Libraries\Search\ScoreSearchParams;
+
+class UserRank
+{
+    public static function getRank(ScoreSearchParams $params): ?int
+    {
+        if ($params->beforeTotalScore === null && $params->beforeScore === null) {
+            return null;
```

Comment:
I can't help but think this should throw instead of returning `null` if not set.